### PR TITLE
Adding DRTs

### PR DESCRIPTION
**Pull request by rchauhan18**
Adding DRTs

### CODE REVIEW

**Review by anjalisheel-wpf — APPROVED** [review]

File: src/Test/Editing/DRT/ReliabilityAssertException.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+namespace DRT
+{
+    // Exception used to test exception continuability.
+    internal class ReliabilityAssertException : Exception
+    {
+        internal ReliabilityAssertException(string description) : base(description)
+        {
+        }
+    }
+}
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/ReliabilityAssertException.cs:16]: add new line at bottom

File: src/Test/Editing/DRT/PasswordBoxSuite.cs [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+        }
+
+        private void SetPassword(string text)
+        {
+            _textbox.Password = text;
+        }
+
+        private PasswordBox _textbox;
+    }
+}
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/PasswordBoxSuite.cs:421]: add new line at bottom

File: src/Test/Editing/DRT/RichTextBoxSuite.cs [hunk 3]
```diff
@@ -0,0 +1,2580 @@
...
+        private FieldInfo _reflectionCaretElement_Top;
+        private FieldInfo _reflectionCaretElement_SystemCaretWidth;
+        private FieldInfo _reflectionCaretElement_InterimWidth;
+        private FieldInfo _reflectionCaretElement_Height;
+        private FieldInfo _reflectionCaretElement_IsBlinkEnabled;
+        private FieldInfo _reflectionCaretElement_CaretBrush;
+        private FieldInfo _reflectionCaretElement_AdornerLayer;
+        private FieldInfo _reflectionCaretElement_Italic;
+    }
+}
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/RichTextBoxSuite.cs:2580]: add new line at bottom

File: src/Test/Editing/DRT/RtfSimple.Rtf [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+\par }\pard \ltrpar\ql \li0\ri0\widctlpar\tx2160\aspalpha\aspnum\faauto\adjustright\rin0\lin0\itap0\pararsid8068530 {\rtlch \af0\afs40 \ltrch \fs40\insrsid8068530 \hich\af0\dbch\af12\loch\f0 T}{\rtlch \af0\afs40 \ltrch \fs40\ul\insrsid8068530\charrsid8732892 
+\hich\af0\dbch\af12\loch\f0 hi}{\rtlch \af0\afs40 \ltrch \fs40\insrsid8068530 \hich\af0\dbch\af12\loch\f0 s }{\rtlch \ab\af0\afs40 \ltrch \b\fs40\insrsid8068530\charrsid8732892 \hich\af0\dbch\af12\loch\f0 is}{\rtlch \af0\afs40 \ltrch \fs40\insrsid8068530 
+\hich\af0\dbch\af12\loch\f0  }{\rtlch \ai\af0\afs40 \ltrch \i\fs40\insrsid8068530\charrsid14775101 \hich\af0\dbch\af12\loch\f0 testing}{\rtlch \af0\afs40 \ltrch \fs40\insrsid8068530 \tab 
+\par }\pard \ltrpar\ql \li0\ri0\widctlpar\aspalpha\aspnum\faauto\adjustright\rin0\lin0\itap0\pararsid14775101 {\rtlch \af0\afs40 \ltrch \fs40\insrsid14775101 
+\par 
+\par }{\rtlch \af0 \ltrch \insrsid14775101\charrsid6104295 
+\par }\pard \ltrpar\ql \li0\ri0\widctlpar\aspalpha\aspnum\faauto\adjustright\rin0\lin0\itap0\pararsid4531514 {\rtlch \ab\af0 \ltrch \b\insrsid14775101\charrsid11547312 
+\par }}
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/RtfSimple.Rtf:97]: add new line at bottom

File: src/Test/Editing/DRT/TextBoxSuite.cs [hunk 5]
```diff
@@ -0,0 +1,1801 @@
...
+
+        protected override IEnumerator LogicalChildren
+        {
+            get
+            {
+                return null;
+            }
+        }
+    }
+}
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/TextBoxSuite.cs:1801]: add new line at bottom

File: src/Test/Editing/DRT/TextChangedSuite.cs [hunk 6]
```diff
@@ -0,0 +1,595 @@
...
+
+            _rtb.EndChange();
+        }
+
+        #endregion Private Methods
+
+        private RichTextBox _rtb;
+        private List<myTextChange> _expectedChanges;
+    }
+}
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/TextChangedSuite.cs:595]: add new line at bottom

File: src/Test/Editing/DRT/TextContainerSuite.cs [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+        void NewHyperlink(TextPointer start, TextPointer end)
+        {
+            // Hyperlink uses a system color.  Override with Blue to make serialization comparision predictable.
+            Hyperlink hl = new Hyperlink(start, end);
+            hl.Foreground = System.Windows.Media.Brushes.Blue;
+        }
+
+        #endregion Private Methods
+    }
+}
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/TextContainerSuite.cs:237]: add new line at bottom

File: src/Test/Editing/DRT/TextElementUndoSuite.cs [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        #endregion Private Methods
+
+        #region Private Fields
+
+        // RichTextBox used by tests.
+        private RichTextBox _richTextBox;
+
+        #endregion Private Fields
+    }
+}
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/TextElementUndoSuite.cs:92]: add new line at bottom

File: src/Test/Editing/DRT/TextRangeSerializationSuite.cs [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+            }
+            set
+            {
+                _customInt = value;
+            }
+        }
+
+        private int _customInt;
+    }
+}
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/TextRangeSerializationSuite.cs:298]: add new line at bottom

File: src/Test/Editing/DRT/TextTreeSuite.cs [hunk 10]
```diff
@@ -0,0 +1,1518 @@
...
+        }
+
+        public MyText(DependencyObject parent) : base(UIContext.CurrentContext)
+        {
+            if (parent != null)
+                ((IAddChild)parent).AddChild(this);
+        }
+    }
+}
+#endif // DISABLED_BY_TOM_BREAKING_CHANGE
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/TextTreeSuite.cs:1518]: add new line at bottom

File: src/Test/Editing/DRT/TypographyFeatures.cs [hunk 11]
```diff
@@ -0,0 +1,1067 @@
...
+        internal string FeatureName;     // four character abbreviation
+        internal int StartCharacter;  // offset of a character combination in source GlyphRun
+        internal int CharacterCount;  // number of unicode charatcres representing a glyph
+        internal int AlternateCount;  // number of typographic alternates available for this glyph with this feature
+        // AlternateCount=1 means that the feature may be a switch or a (single) alternative
+        // AlternateCount>1 means that the feature must be multiple alternative
+
+        #endregion Internal Fields
+    }
+}
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/TypographyFeatures.cs:1067]: add new line at bottom

File: src/Test/Editing/DRT/XamlRtfConverter.cs [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+
+        #endregion Internal Methods
+
+        #region Internal Fields
+
+        internal const int RtfCodePage = 1252;
+
+        #endregion Internal Fields
+    }
+}
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/XamlRtfConverter.cs:88]: add new line at bottom

File: src/Test/Editing/DRT/XamlRtfConverterSuite.cs [hunk 13]
```diff
@@ -0,0 +1,235 @@
...
+        #endregion Private Methods
+
+        #region Private Fields
+
+        private RichTextBox _richTextBox;
+        private TextBox _textBox;
+
+        #endregion Private Fields
+    }
+}
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/XamlRtfConverterSuite.cs:235]: add new line at bottom

File: src/Test/Editing/DRT/input.cs [hunk 14]
```diff
@@ -0,0 +1,469 @@
...
+            {
+                SendKeyboardInput((uint)inputList.Length, inputList, sizeof(KeyboardInput));
+            }
+        }
+
+        #endregion Keyboard support.
+
+        #endregion Input emulation support.
+    }
+}
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/input.cs:469]: add new line at bottom

File: src/Test/Editing/DRT/rtfsimple_expect.rtf.xaml.rtf [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+{\fs40\f0\b\i {\lang1033\ul\cf3\ltrch This}{\lang1033\ul\ltrch  }{\lang1033\ul\fs48\cf2\ltrch i}{\lang1033\loch\f2\ul\fs72\cf7\ltrch T}{\lang1033\ul\fs48\cf4\ltrch s}{\lang1033\ul\ltrch  }{\lang1033\ul\cf8\ltrch t}{\lang1033\ul\cf9\ltrch e}{\lang1033\ul\cf5\ltrch s}{\lang1033\ul\ltrch t}{\lang1033\ul\cf10\ltrch in}{\lang1033\ul\ltrch g.}\li0\ri0\sa0\sb0\fi0\ql\par}
+{\fs40\f0 {\lang1033\ltrch Th}{\lang1033\cf2\ltrch is}{\lang1033\ltrch  }{\lang1033\highlight3\ltrch is}{\lang1033\ltrch  testing}\li0\ri0\sa0\sb0\fi0\ql\par}
+{\fs72\f0 {\lang1033\cf11\ltrch T}{\lang1033\cf3\ltrch h}{\lang1033\cf2\ltrch i}{\lang1033\ltrch s }{\lang1033\fs24\ltrch is}{\lang1033\ltrch  a }{\lang1033\fs48\ltrch test}{\lang1033\fs24\ltrch in}{\lang1033\fs40\ltrch g}\li0\ri0\sa0\sb0\fi0\ql\par}
+{\fs40\f0 {\lang1033\ltrch T}{\lang1033\ul\ltrch hi}{\lang1033\ltrch s }{\lang1033\b\ltrch is}{\lang1033\ltrch  }{\lang1033\i\ltrch testing}{\lang1033\ltrch \tab }\li0\ri0\sa0\sb0\fi0\ql\par}
+{\fs40\f0 \li0\ri0\sa0\sb0\fi0\ql\par}
+{\fs40\f0 \li0\ri0\sa0\sb0\fi0\ql\par}
+{\f0 \li0\ri0\sa0\sb0\fi0\ql\par}
+{\f0\b \li0\ri0\sa0\sb0\fi0\ql\par}
+}
+}
```
Comment (Kuldeep-MS) [inline src/Test/Editing/DRT/rtfsimple_expect.rtf.xaml.rtf:49]: add new line at bottom

File: src/Test/Performance/DRT/MSN.com/MSN.xaml [hunk 16]
```diff
@@ -0,0 +1,875 @@
...
+                    <Canvas Margin="4,0,4,0" />
+                    <Hyperlink NavigateUri="BlankPage.xaml" x:Uid="422">GetNetWise</Hyperlink>
+                </TextBlock>
+                <!-- *** END footer *** -->
+            </DockPanel>
+            <!-- *** END Scrollable panel *** -->
+        </ScrollViewer>
+    </Border>
+</Page>
+<!-- *** END Performance Instrumentation Code -->
```
Comment (Kuldeep-MS) [inline src/Test/Performance/DRT/MSN.com/MSN.xaml:875]: add new line at bottom

File: src/Test/Performance/DRT/MSN.com/MyMSN.xaml [hunk 17]
```diff
@@ -0,0 +1,1218 @@
...
+                        </Hyperlink>
+                    </TextBlock>
+                </DockPanel>
+                <!-- *** END footer *** -->
+            </DockPanel>
+            <!-- *** END Scrollable panel *** -->
+        </ScrollViewer>
+    </Border>
+</Page>
+<!-- *** END Performance Instrumentation Code -->
```
Comment (Kuldeep-MS) [inline src/Test/Performance/DRT/MSN.com/MyMSN.xaml:1218]: add new line at bottom

**Review by Kuldeep-MS — APPROVED** [review]
Overall looks good to me

**Pull request merged**